hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/ConnectionContext.java

```diff
@@ -57,6 +62,17 @@ public synchronized boolean isActive() {
     return this.numThreads > 0;
   }
 
+  /**
+   * Check if the connection is/was active recently.
+   *
+   * @return True if the connection is active or
+   * was active in the past period of time.
+   */
+  public synchronized boolean isActiveRecently() {
+    return isActive() ||
```

Comment:
If we do the timestamp based, do we even need to check for the isActive() or the timestamp comparisson is enoug?

Reply:
That can removed since the timewindow calculation covers the active case. Updated.